smarts/core/agent.py

```diff
@@ -62,87 +56,15 @@ def act(self, obs):
-    """A callable to build an `smarts.core.agent.Agent` given `AgentSpec.agent_params` (default None)"""
-    agent_params: Optional[Any] = None
-    """Parameters to be given to `AgentSpec.agent_builder` (default None)"""
-
-    observation_adapter: Callable = lambda obs: obs
-    """An adaptor that allows shaping of the observations (default lambda obs: obs)"""
-    action_adapter: Callable = lambda act: act
-    """An adaptor that allows shaping of the action (default lambda act: act)"""
-    reward_adapter: Callable = lambda obs, reward: reward
-    """An adaptor that allows shaping of the reward (default lambda obs, reward: reward)"""
-    info_adapter: Callable = lambda obs, reward, info: info
-    """An adaptor that allows shaping of info (default lambda obs, reward, info: info)"""
-
-    def __post_init__(self):
-        # make sure we can pickle ourselves
-        cloudpickle.dumps(self)
-
-    def build_agent(self) -> Agent:
-        """Construct an Agent from the AgentSpec configuration."""
-        if self.agent_builder is None:
-            raise ValueError("Can't build agent, no agent builder was supplied")
-
-        if not callable(self.agent_builder):
-            raise ValueError(
-                f"""agent_builder: {self.agent_builder} is not callable
-Use a combination of agent_params and agent_builder to define how to build your agent, ie.
-
-AgentSpec(
-  agent_params={{"input_dimensions": 12}},
-  agent_builder=MyAgent # we are not instantiating the agent, just passing the class reference
-)
-"""
-            )
-
-        if self.agent_params is None:
-            # no args to agent builder
-            return self.agent_builder()
-        elif isinstance(self.agent_params, (list, tuple)):
-            # a list or tuple is treated as positional arguments
-            return self.agent_builder(*self.agent_params)
-        elif isinstance(self.agent_params, dict):
-            # dictionaries, as keyword arguments
-            fas = inspect.getfullargspec(self.agent_builder)
-            if fas[2] is not None:
-                return self.agent_builder(**self.agent_params)
-            else:
-                return self.agent_builder(
-                    **{
-                        k: self.agent_params[k]
-                        for k in self.agent_params.keys() & set(fas[0])
-                    }
-                )
-        else:
-            # otherwise, the agent params are sent as is to the builder
-            return self.agent_builder(self.agent_params)
+def deprecated_agent_spec(*args, **kwargs):
+    """Deprecated version of AgentSpec, see smarts.zoo.agent_spec"""
+    from smarts.zoo.agent_spec import AgentSpec as AgentSpecAlias
+
+    warnings.warn(
+        "The AgentSpec class has moved to the following module: smarts.zoo.agent_spec. Calling it from this module will be deprecated.",
+        DeprecationWarning,
+    )
+    return AgentSpecAlias(*args, **kwargs)
+
+
+AgentSpec = deprecated_agent_spec
```

Comment:
To explain this for others, this is the heart of this PR. The deprecation method forwards to the `AgentSpec` constructor. The late import is required to avoid circular imports between `core.agent` and `zoo.agent_spec` and is also why inheritance is not used.